inc/Base/Activate.php

```diff
@@ -39,28 +47,46 @@ public static function activate() {
 	 */
 	private static function create_database() {
 		global $wpdb;
-
-		$smaily = "
-                CREATE TABLE IF NOT EXISTS `{$wpdb->prefix}smaily` (
-                `id` int(11) NOT NULL AUTO_INCREMENT ,
-                `enable` tinyint(1) DEFAULT NULL,
-                `subdomain` varchar(255) DEFAULT NULL,
-                `username` varchar(255) DEFAULT NULL,
-                `password` varchar(255) DEFAULT NULL,
-                `autoresponder` varchar(255) DEFAULT NULL,
-                `autoresponder_id` int(10) DEFAULT NULL,
-                `syncronize_additional` varchar(255) DEFAULT NULL,
-                `syncronize` varchar(255) DEFAULT NULL,
-                PRIMARY KEY (`id`)
-                ) ENGINE=InnoDB DEFAULT CHARSET=latin1;
-                ";
-
+		// Create smaily settigs table.
+		$table_name = $wpdb->prefix . 'smaily';
+		$charset_collate = $wpdb->get_charset_collate();
+		$smaily     = "CREATE TABLE $table_name (
+				id int(11) NOT NULL AUTO_INCREMENT,
+				enable tinyint(1) DEFAULT NULL,
+				subdomain varchar(255) DEFAULT NULL,
+				username varchar(255) DEFAULT NULL,
+				password varchar(255) DEFAULT NULL,
+				autoresponder varchar(255) DEFAULT NULL,
+				autoresponder_id int(10) DEFAULT NULL,
+				syncronize_additional varchar(255) DEFAULT NULL,
+				enable_cart tinyint(1) DEFAULT NULL,
+				cart_autoresponder varchar(255) DEFAULT NULL,
+				cart_autoresponder_id int(10) DEFAULT NULL,
+				cart_delay int(10) DEFAULT NULL,
+				cart_cutoff int(10) DEFAULT NULL,
+				cart_options varchar(255) DEFAULT NULL,
+				PRIMARY KEY  (id)
+				) $charset_collate;";
 		dbDelta( $smaily );
 
+		// Create smaily_abandoned_cart table.
+		$abandoned_table_name = $wpdb->prefix . 'smaily_abandoned_carts';
+		$abandoned            = "CREATE TABLE $abandoned_table_name (
+				id int(11) NOT NULL AUTO_INCREMENT,
+				customer_id int(11) DEFAULT NULL,
+				cart_updated datetime DEFAULT '0000-00-00 00:00:00',
+				cart_content longtext DEFAULT NULL,
```

Comment:
Why do we need to store copy of cart contents? Can't cart contents be loaded with `WC()->cart->get_cart()`?

Reply:
I think the `WC()->cart->get_cart()` method works only with customer session. To receive user cart during cron function,  the cart is updated with newest state when ever customer acts on page and `woocommerce_cart_updated` hook is called.

Reply:
OK. Let's keep copy of cart contents.